### ISSUE TITLE
Transition to Chrome new headless mode

### ISSUE THREAD

**Issue by BrendonChau** [quarto-dev/quarto-cli#10532]
Per https://developer.chrome.com/docs/chromium/new-headless
>Note: Passing the `--headless` command-line flag without an explicit value still activates the old Headless mode, but we intend to change this default. We intend to remove the old Headless from the Chrome binary and stop supporting this mode in Puppeteer in 2024. Simultaneously, we'll make old Headless mode available as a standalone binary for users who can't upgrade yet.

**Comment by mcanouil** [comment]
@pmacg Which version of Quarto were you using for this?

**Comment by cderv** [comment]
Wow thanks ! That was faster than I expected !! 

We'll do the necessary update for new chrome or maybe directly use the new headless binary 🤔

**Comment by cderv** [comment]
BTW in the meantime this can help: 

- `QUARTO_CHROMIUM_HEADLESS_MODE=new` can be set so that `--headless=new` is passed. If that does not work 
- `QUARTO_CHROMIUM_HEADLESS_MODE=none` so that `--headless` only is passed. 
- `QUARTO_CHROMIUM` can be used to specified which version of quarto to use.

**Comment by benkeks** [comment]
I ran into the same problem today (using Quarto 1.7.11, Debian Testing, Google Chrome 132.0.6834.83). `QUARTO_CHROMIUM_HEADLESS_MODE=new quarto render --to all` indeed worked as a workaround.

**Comment by cderv** [comment]
> Anyhow, installing 1.7.13 will help I think.

Quarto 1.7.13 is available now.

**Comment by cderv** [comment]
Let's close this one now as starting from 1.7.13, Quarto will use `--headless` as flag by default, so that new headless will be used by default with Chrome from 112. 

Please to report any problem you may encounter with new headless mode if any. Thanks

**Comment by michelk** [comment]
I had the same problem. After updating quarto to 1.7.13, there is another error. 

Any idea, why this is happening?

![Image](https://github.com/user-attachments/assets/475783ae-e12f-4373-a517-12268b637a88)

Export to html works though.

**Comment by mcanouil** [comment]
@michelk Could you open a new issue with all the required information? You can reference this issue in the new one.

Please be aware that screenshots of code do not hold the same value as actual text-based code.
Screenshots are only useful for visual issue. For text, please always prefer using the actual text especially if you don't intend to add alternative text for your screenshots.

**Comment by serkor1** [comment]
Is there a workaround for 1.6.xx? I am running into the same issue as @pmacg on Github actions:

```yaml
ERROR: Couldn't find open server.
Chrome process error: Old Headless mode has been removed from the Chrome binary. Please use the new Headless mode (https://developer.chrome.com/docs/chromium/new-headless) or the chrome-headless-shell which is a standalone implementation of the old Headless mode (https://developer.chrome.com/blog/chrome-headless-shell).



Stack trace:
Chrome process error: Old Headless mode has been removed from the Chrome binary. Please use the new Headless mode (https://developer.chrome.com/docs/chromium/new-headless) or the chrome-headless-shell which is a standalone implementation of the old Headless mode (https://developer.chrome.com/blog/chrome-headless-shell).


    at criClient (file:///opt/quarto/bin/quarto.js:38718:15)
    at eventLoopTick (ext:core/01_core.js:214:9)
    at async file:///opt/quarto/bin/quarto.js:38676:22
    at async Semaphore.runExclusive (file:///opt/quarto/bin/quarto.js:38350:29)
    at async Object.createPngsFromHtml (file:///opt/quarto/bin/quarto.js:38947:42)
    at async makePng (file:///opt/quarto/bin/quarto.js:82312:70)
    at async makeDefault (file:///opt/quarto/bin/quarto.js:82372:24)
    at async Object.cell (file:///opt/quarto/bin/quarto.js:82395:20)
    at async Promise.all (index 0)
    at async Object.document (file:///opt/quarto/bin/quarto.js:39242:25)
Error occurred during cleanup: NotFound: ESRCH: No such process
```

**Comment by gadenbuie** [comment]
You can set the envvar `QUARTO_CHROME_HEADLESS_MODE=new` to use new headless mode with Quarto v1.6.x. Alternatively, to keep using the old headless mode, you could download the [chrome-headless-shell](https://developer.chrome.com/blog/chrome-headless-shell) and then set `QUARTO_CHROMIUM` to the path to the `chrome-headless-shell` binary.

**Comment by serkor1** [comment]
@gadenbuie thank you, but so far no luck. This is my project-structure:

```
Repository/
├─ docs/
│  ├─ _environment
│  ├─ _quarto.yaml
```

Am I supposed to edit the workflow-file too, for it to work? I still get the following error:

```yaml
ERROR: Couldn't find open server.
Chrome process error: Old Headless mode has been removed from the Chrome binary. Please use the new Headless mode (https://developer.chrome.com/docs/chromium/new-headless) or the chrome-headless-shell which is a standalone implementation of the old Headless mode (https://developer.chrome.com/blog/chrome-headless-shell).
```

**Comment by gadenbuie** [comment]
@serkor1 Yeah, I'd try setting this variable in your `build-docs.yaml` workflow:

```patch
    env:
      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      R_KEEP_PKG_SOURCE: yes
+     QUARTO_CHROME_HEADLESS_MODE: new
```

**Comment by serkor1** [comment]
> @serkor1 Yeah, I'd try setting this variable in your `build-docs.yaml` workflow:
> 
> ```patch
>     env:
>       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
>       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
>       R_KEEP_PKG_SOURCE: yes
> +     QUARTO_CHROME_HEADLESS_MODE: new
> ```

I saw the fork 😁 Yeah Ill give it a go!

**Comment by gadenbuie** [comment]
@serkor1 I hope it helps! If not, it'd be helpful to extract a smaller reproducible example that demonstrates the problem you're encountering. It's very hard to debug in a large repository with so many moving pieces.

**Comment by serkor1** [comment]
> @serkor1 I hope it helps! If not, it'd be helpful to extract a smaller reproducible example that demonstrates the problem you're encountering. It's very hard to debug in a large repository with so many moving pieces.

Ofcourse, many variables to consider. Ill make a mock repo sometime soon!

But thank you so much!

**Comment by mcanouil** [comment]
May the next messages be on the discussion that has been opened.
The issue here has been closed and this make very hard to follow two parallel discussions on the same problem.

- https://github.com/quarto-dev/quarto-cli/discussions/11966
